### DISCUSSION TITLE
Auth by key pair & signature

### DISCUSSION THREAD

**Discussion by anotherhadi** [pocketbase/pocketbase#2843] [Q&A] [answered]
Hi, I'm creating an app and I was wondering if Pocketbase was a good choice to auth users with a key pair.

I'll need to create keys when they sign up and check for the signature on some endpoints, is this possible with Pocketbase?

**Comment by ganigeorgiev** [comment, marked as answer]
Could you elaborate a little more on your use case and why the regular PocketBase JWT auth will not suffice?

We don't have a generic way to verify a key signature as it would deffer on what actually is the key and how it was generated. When [using as framework](https://pocketbase.io/docs/use-as-framework/), you can [register your own custom middleware](https://pocketbase.io/docs/router/#middlewares) and perform the check from there.

In general, if the key is a static value and you want to compare it against some stored user field, you can pass it as a header value with the request and use it in the API rules like `@request.headers.x_key = user.key` (_note that the JS SDK currently doesn't have option to pass custom headers per request basis, only globally; this will be fixed with the planned refactoring_).
